### DISCUSSION TITLE
Multiple output types

### DISCUSSION THREAD

**Discussion by jayvdb** [lowlighter/metrics#602] [Ideas]
It would be nice to be able to emit the json, and the svg, and possibly other types (including the special 'insights'), in a single job.

Or the json might be seen as a cache (similar to the markdown cache) and it can be emitted in addition to the normal output format.

A json cache could be a way to allow the data collection to do 'in depth' across longer periods of time, by slowing collating from more and more repos in each invocation, by trusting the previous json stats for repos already analysed within a specified period.

**Comment by lowlighter** [comment]
Outputting multiple format in a single job could be done but it would be restricted to a same config (i.e. markdown/pdf would be incompatible with svg/png/jpeg and you cannot change plugins in the same job). The 'insights' output is also incompatible with other since it's computed in a quite different way.
I'm not sure this bring much value, as this is something that could be handled directly by github actions directly using a `strategy`.

However I actually like the idea of some sort of cache for data heavy metrics. The only issue is how to keep "integrity" ?
Using external files directly hosted on user repo makes them easily temperable/editable 🤔

**Reply by jayvdb** [reply]
Regarding integrity, it boils down to the same problem you mentioned in https://github.com/lowlighter/metrics/discussions/592#discussioncomment-1526560
> .. this action code can be forked and changed easily

One approach is for peers to verify the stats (JSON) of others while computing their own, cross-checking.  e.g. @gaocegege and I have worked together on a set of repos.  My action worker could compute and verify the overlapping parts of @gaocegege's JSON if was  granular to repo level, with a bit of fuzz allowed, and vis-versa, and it wouldn't take too much extra compute/time.  Or I could periodically do a full compute of @gaocegege's JSON in my repos CI, and emit a file into the repo which lists the other users my CI has checked and when, and he could include a "check here to verify link" on his README.md .  Could even extend it further to be a referee network of sorts, as I would be happy to take calls about his work generally beyond just the metrics.
